python-package-manager/install-dependencies/poetry/action.yml

```diff
@@ -0,0 +1,38 @@
+name: Install poetry dependencies
+inputs:
+  python-version:
+    default: "3.11"
+    required: false
+    type: string
+  requirements-files:
+    default: requirements.txt
+    description: Space-separated list of requirements files if using pip
+    required: false
+    type: string
+  use-private-repo:
+    default: false
+    description: Use the ZUoS private GCloud Artifact Registry (required GCloud authentication)
+    required: false
+    type: boolean
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install poetry
+      run: pipx install poetry
+      shell: bash
+    - name: Ensure virtualenv is not created in the project  # so that setup-python cache will work
+      run: rm -rf .venv
```

Comment:
You don't want to install the poetry virtual environment in the `.venv` directory but in `.local/share/virtualenv/pypoetry/` instead? Either way is fine. Only when building a container, being able to copy the full local directory with the `.venv` directory in it might be easier, since poetry automatically picks up that directory if it exists. But if `.venv` doesn't exist, it will use `~/.local/share/virtualenv/pypoetry` by default.

Reply:
But I don't have a preference, as long as its not an oversight.

Reply:
> If there is a `.venv` directory present, it will use it by default as well.

Yes but setting the setting means we don't need to version control a local `.venv` or remove it in the CI steps, both "_just work_" from the setting.

We need the local `.venv` for the `pyright` pre-commit hook to get dependencies when it runs, but other than that it doesn't matter where the venv goes.

Reply:
Good point.